app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt

```diff
@@ -12,45 +12,119 @@ import androidx.compose.foundation.clickable
 import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import kotlinx.coroutines.launch
 import org.mozilla.fenix.R
 import org.mozilla.fenix.theme.FirefoxTheme
 import org.mozilla.fenix.wallpapers.Wallpaper
 
 /**
- * The screen for controlling settings around Wallpapers.
+ * The screen for controlling settings around Wallpapers. When a new wallpaper is selected,
+ * a snackbar will be displayed.
  *
  * @param wallpapers Wallpapers to add to grid.
  * @param selectedWallpaper The currently selected wallpaper.
- * @param onSelectWallpaper Action to take when a new wallpaper is selected.
+ * @param onSelectWallpaper Callback for when a new wallpaper is selected.
+ * @param onViewWallpaper Callback for when the view action is clicked from snackbar.
  */
 @Composable
 fun WallpaperSettings(
     wallpapers: List<Wallpaper>,
     selectedWallpaper: Wallpaper,
     onSelectWallpaper: (Wallpaper) -> Unit,
+    onViewWallpaper: () -> Unit,
 ) {
-    Surface(color = FirefoxTheme.colors.layer2) {
-        WallpaperThumbnails(
-            wallpapers = wallpapers,
-            selectedWallpaper = selectedWallpaper,
-            onSelectWallpaper = onSelectWallpaper
-        )
+    val coroutineScope = rememberCoroutineScope()
+    val scaffoldState = rememberScaffoldState()
+
+    Scaffold(
+        backgroundColor = FirefoxTheme.colors.layer1,
+        scaffoldState = scaffoldState,
+        snackbarHost = { hostState ->
+            SnackbarHost(hostState = hostState) {
+                WallpaperSnackbar(onViewWallpaper)
+            }
+        }
+    ) {
+        Surface(color = FirefoxTheme.colors.layer2) {
+            WallpaperThumbnails(
+                wallpapers = wallpapers,
+                selectedWallpaper = selectedWallpaper,
+                onSelectWallpaper = { updatedWallpaper ->
+                    coroutineScope.launch {
+                        scaffoldState.snackbarHostState.showSnackbar(
+                            message = "", // overwritten by WallpaperSnackbar
+                            duration = SnackbarDuration.Short
+                        )
+                    }
+                    onSelectWallpaper(updatedWallpaper)
+                },
+            )
+        }
     }
 }
 
+@Composable
+private fun WallpaperSnackbar(
+    onViewWallpaper: () -> Unit,
+) {
+    Snackbar(
+        modifier = Modifier
+            .padding(8.dp)
+            .heightIn(min = 48.dp),
+        backgroundColor = FirefoxTheme.colors.actionPrimary,
+        content = {
+            Text(
+                modifier = Modifier.padding(start = 16.dp, end = 16.dp, top = 8.dp, bottom = 8.dp),
```

Comment:
Looking at the @Preview, it seems like the Snackbar has some built-in padding already so we might not actually need this, but let's look at this in a followup.